### PR TITLE
Data widgets module

### PR DESCRIPTION
**Pull request by diego-antonelli**
Introducing documentation for Data widgets module.

This module includes the content of Data Grid 2, Gallery, Filters, Sort and Tree node.

It also merges the following PRs:
- https://github.com/mendix/docs/pull/3674
- https://github.com/mendix/docs/pull/3648

@ConnorLand Gallery docs contains your latest changes.

@Luyao-Zhang-1 Tree node docs also contains your latest changes.

### CODE REVIEW

**Comment by Keraito** [comment]
For visibility: 

I reviewed the tree node docs again per @Luyao-Zhang-1 's request. Some small feedback that I also DMd her:
- Line 59: not only the header, also the content. But not sure if that’s worth mentioning (here)
- Line 64: we changed hasChildren default to Yes. This will make it clear that users can nest tree nodes.
- Line 74: the advanced toggle will only be available in Studio. So in Studio Pro the “Advanced Options” property will always be “Yes” and it will have all features (this change will apply starting 9.6)

File: content/appstore/modules/data-grid-2.md [hunk 1]
```diff
@@ -322,7 +344,7 @@ Here is an example of a configuration containing an On change action:
 
 ## 9 Performance
 
-The performance of the new data grid can be affected if sorting or filtering are enabled. This is because Data Grid 2 v1.1.0 retrieves all available values to do the sorting or filtering while the widget runs.
+The performance of the new data grid can be affected if sorting or filtering are enabled. This is because Data Grid 2 and Data widgets v1.1 - 1.2 retrieve all available values for sorting or filtering while the widget runs. This problem does not occur in versions higher than 1.3 because those version use [Filtering Apis](/apidocs-mxsdk/apidocs/pluggable-widgets-client-apis-list-values#4-filter-helpers) to retrieve only the necessary data.
```
Comment (diego-antonelli) [inline content/appstore/modules/data-grid-2.md:347]: @ConnorLand we are gonna rename Data Grid 2 module to Data Widgets, thats why i added a slash in the docs.

Comment (ConnorLand) [reply content/appstore/modules/data-grid-2.md:347]: Oh excuse me, I understand now. Well, that rename is going to be final when this PR is merged leaving only "Data Widgets" -- no more Data Grid 2. Right? So instead should I change this to only Data Widgets?

Comment (diego-antonelli) [reply content/appstore/modules/data-grid-2.md:347]: Thats indeed. So lets change it to Data Widgets only.

**Comment by Luyao-Zhang-1** [comment]
@Keraito I made the [changes](https://github.com/mendix/docs/pull/3698/commits/b8b0c9bf0e720ad5b1762258b017718257f9f658) based on your feedback. Please take a look.

File: content/appstore/modules/data-grid-2.md (hunk 1 above)
Comment (ConnorLand) [reply content/appstore/modules/data-grid-2.md]: I updated this doc from Data Grid 2 to Data Widgets. Can you also do a find + replace in your PR for other instances of Data Grid 2 and commit?

Comment (diego-antonelli) [reply content/appstore/modules/data-grid-2.md]: I will update in the workflow commons as there they mention Data Grid 2.

File: content/appstore/modules/data-grid-2.md [hunk 2]
```diff
@@ -1,20 +1,21 @@
 ---
-title: "Data Grid 2"
+title: "Data Widgets"
```
Comment (diego-antonelli) [inline content/appstore/modules/data-grid-2.md:2]: This document should not be renamed @ConnorLand. We will still have Data Grid 2, but is part of the Data module.

File: content/appstore/modules/data-grid-2.md (hunk 1 above)
Comment (diego-antonelli) [reply content/appstore/modules/data-grid-2.md]: @ConnorLand did that in my last commit. Can you check?

Comment (ConnorLand) [reply content/appstore/modules/data-grid-2.md]: Made some small changes. Looks good.

**Review by ConnorLand — APPROVED** [review]

**Pull request merged**